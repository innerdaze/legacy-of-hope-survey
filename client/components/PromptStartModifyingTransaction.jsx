import React, { Component } from 'react'
import Layer from 'grommet/components/Layer'
import Header from 'grommet/components/Header'
import Heading from 'grommet/components/Heading'
import Title from 'grommet/components/Title'
import Paragraph from 'grommet/components/Paragraph'
import Footer from 'grommet/components/Footer'
import Anchor from 'grommet/components/Anchor'
import CloseIcon from 'grommet/components/icons/base/Close'
import NextLinkIcon from 'grommet/components/icons/base/LinkNext'

export default ({
  onSubmit = Function.prototype,
  onCancel = Function.prototype
}) => (
  <Layer>
    <Header>
      <Title>
        Not so fast!
      </Title>
    </Header>
    <Heading tag='h2'>
      There is already an entry for this product.
    </Heading>
    <Paragraph>
      Would you like to change the quantity?
    </Paragraph>
    <Footer
      pad={{ vertical: 'medium' }}
      justify='between'
      >
      <Anchor
        label='Cancel'
        icon={<CloseIcon/>}
        onClick={onCancel}
        />
      <Anchor
        primary
        label='Continue'
        icon={<NextLinkIcon/>}
        onClick={onSubmit}
        />
    </Footer>
  </Layer>
)
