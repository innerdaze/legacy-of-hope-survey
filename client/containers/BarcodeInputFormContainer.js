import { connect } from 'react-redux'
import BarcodeInputForm from '../components/BarcodeInputForm'
import { orderOperations, orderSelectors } from '../features/order'
import { barcodeSelectors } from '../features/barcode'
import { syncSelectors } from '../features/sync'
import { uiSelectors } from '../features/ui'
import { wastageOperations } from '../features/wastage'
const mapStateToProps = state => {
  return {
    error: barcodeSelectors.lastError(state),
    orderMode: orderSelectors.modeSelector(state),
    shouldFocusField:
      !uiSelectors.mainMenuVisible(state) &&
      !orderSelectors.isChangingOrderQuantitySelector(state) &&
      !orderSelectors.pendingModificationSelector(state) &&
      !orderSelectors.isProcessingSelector(state) &&
      !orderSelectors.pendingTransactionSelector(state) &&
      !orderSelectors.isDeletingOrderSelector(state) &&
      !syncSelectors.isSyncing(state)
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onSubmitBarcode: barcodeID =>
      dispatch(orderOperations.submitBarcode(barcodeID))
  }
}

const BarcodeInputFormContainer = connect(mapStateToProps, mapDispatchToProps)(
  BarcodeInputForm
)

export default BarcodeInputFormContainer
