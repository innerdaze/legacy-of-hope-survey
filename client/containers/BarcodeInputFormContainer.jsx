import { connect } from 'react-redux'
import BarcodeInputForm from '../components/BarcodeInputForm'
import { createPendingTransactionByBarcodeID } from '../actions/OrderActions'

const mapStateToProps = state => {
  return {
    error: state.barcodeLookup.lastError,
    shouldFocusField: (
        !state.ui.mainMenuVisible &&
        !state.orders.isChangingOrderQuantity &&
        !state.orders.pendingModification &&
        !state.orders.isProcessing &&
        !state.orders.pendingTransaction &&
        !state.orders.isDeletingOrder &&
        !state.sync.isSyncing
    )
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onSubmitBarcode: barcodeID => (
      dispatch(createPendingTransactionByBarcodeID(barcodeID))
    )
  }
}

const BarcodeInputFormContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(BarcodeInputForm)

export default BarcodeInputFormContainer
