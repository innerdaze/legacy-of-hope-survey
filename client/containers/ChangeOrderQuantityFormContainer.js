import { connect } from 'react-redux'
import { orderOperations } from '../features/order'
import { inventorySelectors } from '../features/inventory'
import ChangeOrderQuantityForm from '../components/ChangeOrderQuantityForm'

const mapStateToProps = state => {
  return {
    order: inventorySelectors.currentlyChangeQuantityForWithProductSelector(
      state
    )
  }
}

const mapDispatchToProps = dispatch => {
  return {
    handleSubmit: (id, quantity) => {
      dispatch(orderOperations.changeOrderQuantity(id, quantity))
      dispatch(orderOperations.finishChangingOrderQuantity())
    },
    handleCancel: () => {
      dispatch(orderOperations.cancelChangingOrderQuantity())
      dispatch(orderOperations.finishChangingOrderQuantity())
    }
  }
}

const ChangeOrderQuantityFormContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ChangeOrderQuantityForm)

export default ChangeOrderQuantityFormContainer
