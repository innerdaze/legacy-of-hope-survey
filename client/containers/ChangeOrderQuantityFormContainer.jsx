import { connect } from 'react-redux'
import { changeOrderQuantity } from '../actions/OrderActions'
import ChangeOrderQuantityForm from '../components/ChangeOrderQuantityForm.jsx'

const mapStateToProps = state => {
  return {}
}

const mapDispatchToProps = dispatch => {
  return {
    onSubmit: (id, quantity) => {
      dispatch(changeOrderQuantity(id, quantity))
    }
  }
}

const ChangeOrderQuantityFormContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ChangeOrderQuantityForm)

export default ChangeOrderQuantityFormContainer
