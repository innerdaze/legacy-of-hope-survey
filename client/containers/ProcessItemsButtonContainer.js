import { connect } from 'react-redux'
import { orderOperations, orderSelectors } from '../features/order'
import ProcessItemsButton from '../components/ProcessItemsButton'

const mapStateToProps = state => {
  return {
    mode: orderSelectors.modeSelector(state)
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onProcessItemsClick: () => {
      dispatch(orderOperations.processOrders())
    }
  }
}

const ProcessItemsButtonContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ProcessItemsButton)

export default ProcessItemsButtonContainer
