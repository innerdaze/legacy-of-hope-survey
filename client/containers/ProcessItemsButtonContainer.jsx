import { connect } from 'react-redux'
import { processOrders } from '../actions/OrderActions'
import ProcessItemsButton from '../components/ProcessItemsButton'

const mapStateToProps = state => {
  return {
    mode: state.orders.mode
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onProcessItemsClick: () => {
      dispatch(processOrders())
    }
  }
}

const ProcessItemsButtonContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ProcessItemsButton)

export default ProcessItemsButtonContainer
