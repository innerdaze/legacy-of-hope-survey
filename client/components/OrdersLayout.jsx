import React, { Component } from 'react'
import BackgroundSyncProgressContainer from '../containers/BackgroundSyncProgressContainer.jsx'
import BarcodeInputFormContainer from '../containers/BarcodeInputFormContainer.jsx'
import ScannedItemListContainer from '../containers/ScannedItemListContainer.jsx'
import ModeSwitcherContainer from '../containers/ModeSwitcherContainer.jsx'
import ProcessItemsButtonContainer from '../containers/ProcessItemsButtonContainer.jsx'
import Box from 'grommet/components/Box'
import Section from 'grommet/components/Section'
import Footer from 'grommet/components/Footer'

const OrdersLayout = () => (
  <Section
    primary={true}
    appCentered={true}
    direction='column'
    full={true}>
    <BackgroundSyncProgressContainer/>
    <ModeSwitcherContainer/>
    <BarcodeInputFormContainer/>
    <ScannedItemListContainer/>
    <ProcessItemsButtonContainer/>
  </Section>
)

export default OrdersLayout
