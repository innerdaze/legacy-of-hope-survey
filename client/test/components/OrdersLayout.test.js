import React from 'react'
import OrdersLayout from '../../components/OrdersLayout'
import BarcodeInputFormContainer from '../../containers/BarcodeInputFormContainer'
import ScannedItemListContainer from '../../containers/ScannedItemListContainer'
import ProcessItemsButtonContainer from '../../containers/ProcessItemsButtonContainer'
import PromptStartModifyingTransaction from '../../components/PromptStartModifyingTransaction'
import ChangeOrderQuantityFormContainer from '../../containers/ChangeOrderQuantityFormContainer'

let wrapper,
  mainMenuVisible,
  pendingModification,
  isChangingOrderQuantity,
  onPromptStartModifyingSubmit,
  onPromptStartModifyingCancel

describe('Testing on OrdersLayout...', () => {
  beforeEach(() => {
    mainMenuVisible = false
    pendingModification = null
    isChangingOrderQuantity = false
    onPromptStartModifyingSubmit = jest.fn()
    onPromptStartModifyingCancel = jest.fn()

    wrapper = shallow(
      <OrdersLayout
        mainMenuVisible={mainMenuVisible}
        pendingModification={pendingModification}
        isChangingOrderQuantity={isChangingOrderQuantity}
        onPromptStartModifyingSubmit={onPromptStartModifyingSubmit}
        onPromptStartModifyingCancel={onPromptStartModifyingCancel}
      />
    )
  })

  describe('OrdersLayout layout', () => {
    test('Expect OrdersLayout to have a Split comp with a MainMenu and a Box comp in it', () => {
      expect(wrapper.find('Split')).toHaveLength(1)
      expect(wrapper.find('Split').find('MainMenu')).toHaveLength(1)
      expect(wrapper.find('Split').find('Box')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })

    test('Expect Box contains OrdersHeaderLayout, BarcodeInputFormContainer, ScannedItemListContainer and ProcessItemsButtonContainer by default', () => {
      expect(wrapper.find('Box').find('OrdersHeaderLayout')).toHaveLength(1)
      expect(
        wrapper.find('Box').contains(<BarcodeInputFormContainer />)
      ).toEqual(true)
      expect(
        wrapper.find('Box').contains(<ScannedItemListContainer />)
      ).toEqual(true)
      expect(
        wrapper.find('Box').contains(<ProcessItemsButtonContainer />)
      ).toEqual(true)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect Split priority prop is set with left if mainMenuVisible prop is true', () => {
      mainMenuVisible = true

      wrapper = shallow(
        <OrdersLayout
          mainMenuVisible={mainMenuVisible}
          pendingModification={pendingModification}
          isChangingOrderQuantity={isChangingOrderQuantity}
          onPromptStartModifyingSubmit={onPromptStartModifyingSubmit}
          onPromptStartModifyingCancel={onPromptStartModifyingCancel}
        />
      )

      expect(wrapper.prop('priority')).toEqual('left')
      expect(wrapper).toMatchSnapshot()
    })

    test('Expect Split priority prop is set with left if mainMenuVisible is not passed', () => {
      wrapper = shallow(
        <OrdersLayout
          pendingModification={pendingModification}
          isChangingOrderQuantity={isChangingOrderQuantity}
          onPromptStartModifyingSubmit={onPromptStartModifyingSubmit}
          onPromptStartModifyingCancel={onPromptStartModifyingCancel}
        />
      )

      expect(wrapper.prop('priority')).toEqual('right')
      expect(wrapper).toMatchSnapshot()
    })

    test('Expect to contain PromptStartModifyingTransaction if pendingModification is not a null object', () => {
      pendingModification = {}

      wrapper = shallow(
        <OrdersLayout
          pendingModification={pendingModification}
          isChangingOrderQuantity={isChangingOrderQuantity}
          onPromptStartModifyingSubmit={onPromptStartModifyingSubmit}
          onPromptStartModifyingCancel={onPromptStartModifyingCancel}
        />
      )

      expect(wrapper.find('PromptStartModifyingTransaction')).toHaveLength(1)
      expect(wrapper).toMatchSnapshot()
    })

    test('Expect to contain ChangeOrderQuantityFormContainer if isChangingOrderQuantity is true', () => {
      isChangingOrderQuantity = true

      wrapper = shallow(
        <OrdersLayout
          pendingModification={pendingModification}
          isChangingOrderQuantity={isChangingOrderQuantity}
          onPromptStartModifyingSubmit={onPromptStartModifyingSubmit}
          onPromptStartModifyingCancel={onPromptStartModifyingCancel}
        />
      )

      expect(wrapper.contains(<ChangeOrderQuantityFormContainer />)).toEqual(
        true
      )
      expect(wrapper).toMatchSnapshot()
    })

    test('Expect handlePromptStartModifyingSubmit prop function is called is PromptStartModifyingTransaction onCancel is trigger', () => {
      pendingModification = {}

      wrapper = shallow(
        <OrdersLayout
          pendingModification={pendingModification}
          isChangingOrderQuantity={isChangingOrderQuantity}
          onPromptStartModifyingSubmit={onPromptStartModifyingSubmit}
          onPromptStartModifyingCancel={onPromptStartModifyingCancel}
        />
      )

      wrapper.find('PromptStartModifyingTransaction').prop('onSubmit')()
      expect(onPromptStartModifyingSubmit).toHaveBeenCalled()
      expect(wrapper).toMatchSnapshot()
    })

    test('Expect onPromptStartModifyingCancel prop function is called is PromptStartModifyingTransaction onCancel is trigger', () => {
      pendingModification = {}

      wrapper = shallow(
        <OrdersLayout
          pendingModification={pendingModification}
          isChangingOrderQuantity={isChangingOrderQuantity}
          onPromptStartModifyingSubmit={onPromptStartModifyingSubmit}
          onPromptStartModifyingCancel={onPromptStartModifyingCancel}
        />
      )

      wrapper.find('PromptStartModifyingTransaction').prop('onCancel')()
      expect(onPromptStartModifyingCancel).toHaveBeenCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
