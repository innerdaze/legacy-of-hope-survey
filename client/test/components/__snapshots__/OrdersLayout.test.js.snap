// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on OrdersLayout... OrdersLayout layout Expect Box contains OrdersHeaderLayout, BarcodeInputFormContainer, ScannedItemListContainer and ProcessItemsButtonContainer by default 1`] = `
<Split
  fixed={true}
  flex="right"
  priority="right"
  showOnResponsive="priority"
>
  <MainMenu />
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="medium"
    responsive={true}
    tag="div"
  >
    <OrdersHeaderLayout />
    <Connect(BarcodeInputForm) />
    <Connect(ScannedItemList) />
    <Connect(ProcessItemsButton) />
  </Box>
</Split>
`;

exports[`Testing on OrdersLayout... OrdersLayout layout Expect OrdersLayout to have a Split comp with a MainMenu and a Box comp in it 1`] = `
<Split
  fixed={true}
  flex="right"
  priority="right"
  showOnResponsive="priority"
>
  <MainMenu />
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="medium"
    responsive={true}
    tag="div"
  >
    <OrdersHeaderLayout />
    <Connect(BarcodeInputForm) />
    <Connect(ScannedItemList)
      renderItem={[Function]}
    />
    <Connect(ProcessItemsButton) />
  </Box>
</Split>
`;

exports[`Testing on OrdersLayout... Test on prop Expect Split priority prop is set with left if mainMenuVisible is not passed 1`] = `
<Split
  fixed={true}
  flex="right"
  priority="right"
  showOnResponsive="priority"
>
  <MainMenu />
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="medium"
    responsive={true}
    tag="div"
  >
    <OrdersHeaderLayout />
    <Connect(BarcodeInputForm) />
    <Connect(ScannedItemList)
      renderItem={[Function]}
    />
    <Connect(ProcessItemsButton) />
  </Box>
</Split>
`;

exports[`Testing on OrdersLayout... Test on prop Expect Split priority prop is set with left if mainMenuVisible prop is true 1`] = `
<Split
  fixed={true}
  flex="right"
  priority="left"
  showOnResponsive="priority"
>
  <MainMenu />
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="medium"
    responsive={true}
    tag="div"
  >
    <OrdersHeaderLayout />
    <Connect(BarcodeInputForm) />
    <Connect(ScannedItemList)
      renderItem={[Function]}
    />
    <Connect(ProcessItemsButton) />
  </Box>
</Split>
`;

exports[`Testing on OrdersLayout... Test on prop Expect handlePromptStartModifyingSubmit prop function is called is PromptStartModifyingTransaction onCancel is trigger 1`] = `
<Split
  fixed={true}
  flex="right"
  priority="right"
  showOnResponsive="priority"
>
  <MainMenu />
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="medium"
    responsive={true}
    tag="div"
  >
    <PromptStartModifyingTransaction
      onCancel={[Function]}
      onSubmit={[Function]}
      order={Object {}}
    />
    <OrdersHeaderLayout />
    <Connect(BarcodeInputForm) />
    <Connect(ScannedItemList)
      renderItem={[Function]}
    />
    <Connect(ProcessItemsButton) />
  </Box>
</Split>
`;

exports[`Testing on OrdersLayout... Test on prop Expect onPromptStartModifyingCancel prop function is called is PromptStartModifyingTransaction onCancel is trigger 1`] = `
<Split
  fixed={true}
  flex="right"
  priority="right"
  showOnResponsive="priority"
>
  <MainMenu />
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="medium"
    responsive={true}
    tag="div"
  >
    <PromptStartModifyingTransaction
      onCancel={[Function]}
      onSubmit={[Function]}
      order={Object {}}
    />
    <OrdersHeaderLayout />
    <Connect(BarcodeInputForm) />
    <Connect(ScannedItemList)
      renderItem={[Function]}
    />
    <Connect(ProcessItemsButton) />
  </Box>
</Split>
`;

exports[`Testing on OrdersLayout... Test on prop Expect to contain ChangeOrderQuantityFormContainer if isChangingOrderQuantity is true 1`] = `
<Split
  fixed={true}
  flex="right"
  priority="right"
  showOnResponsive="priority"
>
  <MainMenu />
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="medium"
    responsive={true}
    tag="div"
  >
    <Connect(ChangeOrderQuantityForm) />
    <OrdersHeaderLayout />
    <Connect(BarcodeInputForm) />
    <Connect(ScannedItemList)
      renderItem={[Function]}
    />
    <Connect(ProcessItemsButton) />
  </Box>
</Split>
`;

exports[`Testing on OrdersLayout... Test on prop Expect to contain PromptStartModifyingTransaction if pendingModification is not a null object 1`] = `
<Split
  fixed={true}
  flex="right"
  priority="right"
  showOnResponsive="priority"
>
  <MainMenu />
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad="medium"
    responsive={true}
    tag="div"
  >
    <PromptStartModifyingTransaction
      onCancel={[Function]}
      onSubmit={[Function]}
      order={Object {}}
    />
    <OrdersHeaderLayout />
    <Connect(BarcodeInputForm) />
    <Connect(ScannedItemList)
      renderItem={[Function]}
    />
    <Connect(ProcessItemsButton) />
  </Box>
</Split>
`;
